### PR TITLE
Add AppLockSetupPINScreen.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR creates the AppLockSetupPINScreen. Not connected to a flow yet.

test_appLockSetupPINScreen Create

### CODE REVIEW

File: ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    var isSecure = false
+    
+    var body: some View {
+        textField
+            .textFieldStyle(PINTextFieldStyle(pinCode: pinCode, isSecure: isSecure))
+            .keyboardType(.numberPad)
+            .onChange(of: pinCode) { newValue in
+                // add some tests.
+                let sanitized = sanitize(newValue)
+                if sanitized != newValue { pinCode = sanitized }
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift:32]: Is this really necessary, would it trigger a change if they're equal?

File: ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    @ViewBuilder
+    var textField: some View {
+        if isSecure {
+            SecureField("", text: $pinCode)
+        } else {
+            TextField("", text: $pinCode)
+        }
+    }
+    
+    func sanitize(_ pinCode: String) -> String {
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift:45]: This reminds me, should we disable pasting on this text field?

File: ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+struct PINTextField: View {
+    @Binding var pinCode: String
+    var isSecure = false
+    
+    var body: some View {
+        textField
+            .textFieldStyle(PINTextFieldStyle(pinCode: pinCode, isSecure: isSecure))
+            .keyboardType(.numberPad)
+            .onChange(of: pinCode) { newValue in
+                // add some tests.
```
[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift:30]: Oh yeah, already done 😁

File: ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift:45]: Its not possible to paste via the long press menu. I hadn't thought about cmd + c / v until just now. Seems it isn't possible in SwiftUI, you need a custom `UITextView` which is probably overkill here. I think I'm happy sanitising the input if someone does try to paste with a shortcut.

File: ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/AppLock/AppLockSetupPINScreen/View/PINTextField.swift:32]: Well it *would* recompute the view needlessly if they're equal but wouldn't trigger a change in here. I think its fine to leave as is. I might add a warning log though that invalid characters were detected.

**Review by stefanceriu — APPROVED** [review]

**Pull request merged**